### ISSUE TITLE
When using New-NsxFirewallRule -AppliedTo not working correctly

### ISSUE THREAD

**Issue by kevsterd** [vmware/powernsx#429]
When using New-NsxFirewallRule to add a new rule to the DFW and the -AppliedTo method to specify ESG's, the rule is added with the correct targets but also includes Distributed Firewall.  

Whilst the distributed firewall should be the default, when adding targets its should be removed.

**Comment by kevsterd** [comment]
Anyone tried this or have any comments ?

**Comment by dcoghlan** [comment]
Hi @kevsterd 

What you are seeing is a valid configuration from the point of view of NSX. However to achieve what you are looking for (only apply the rule to the edge specified) you can do the following:
```
PS> $section = new-nsxfirewallsection -Name AppliedToTest
PS> Get-NsxFirewallSection -objectId $section.id | New-NsxFirewallRule -Name TestAppliedTo1 -Action
 deny -ApplyToDfw:$false -AppliedTo (Get-NsxEdge | select -First 1)

id            : 1006
disabled      : false
logged        : false
name          : TestAppliedTo1
action        : deny
appliedToList : appliedToList
sectionId     : 1004
direction     : inout
packetType    : any

PS> (get-nsxfirewallrule -RuleId 1006).appliedtolist.appliedto

name                          value                         type                          isValid
----                          -----                         ----                          -------
InternetFW                    edge-1                        Edge                          true
```

**Comment by kevsterd** [comment]
I don't agree.   Whist adding a new rule in the GUI does indeed first apply to DFW only.  You then have to delete and set the ESG as target before applying.

If you look at the logic of the PowerNSX function it trys to apply the correct definition if you select the -appliedto option.  However something fails in the logic which causes it to fail.   We have got around this by changing the default action within New-NsxFirewallRule of $ApplyToDFW to false, which is a poor way of doing it.

I cant see a valid use of apply to DFW and an ESG at the same time, apart from remote ESG's

**Comment by nmbradford** [comment]
@kevsterd, what @dcoghlan has noted is correct.  The behaviour of PowerNSX is different to the UI in this instance, but you simply have to explicitly negate the -ApplyToDFW switch as he indicated to get the desired outcome.  You do not need to modify the function as you seem to be implying above. 
 For the sake of clarity, simply negate the -ApplyToDFW switch as follows:

```New-NsxFirewallRule -Name TestAppliedTo1 -Action deny -ApplyToDfw:$false -AppliedTo $nsxedge```

Having said all that, my goal has always been to have consistency between the UI and PowerNSX behaviour, and I agree with you, this is a counter intuitive (though perfectly valid) configuration.

I've taken a look at the function that builds the Applied to XML and to fix this is trivial.  My only hesitation is that it is a breaking change that may impact existing scripts that rely on this behaviour.  I also agree that there arent too many valid uses cases for this functionality, but breaking changes are something we work hard to avoid.  I actually recall not 'fixing' this some time ago for precisely this reason.

I propose to add a deprecation warning that will indicate that this behaviour will change in a future version of PowerNSX and change the behaviour in the next major version.

Lastly - Im confused by your comment 'If you look at the logic of the PowerNSX function it trys to apply the correct definition if you select the -appliedto option. However something fails in the logic which causes it to fail. ' - specifically, the behaviour, and how to work around it is expected and nothing 'fails'.  Am I missing something?

**Comment by kevsterd** [comment]
@nmbradford thanks for the reply.   That makes sense now.   I'd hadn't got that you can include a switch but then set it to false.   Ignore my comments about the code flow, as I'm still getting my head some aspects of how PowerNSX modules are coded.

Thanks for making the aspects clear.   PowerNSX is a great tool and any enhancements that can be made are welcome.

**Comment by alagoutte** [comment]
Yes, with powershell, you need to known howto set a negative switch... (not really natural..)
